src/EnergyPlus/HybridEvapCoolingModel.cc

```diff
@@ -0,0 +1,2383 @@
+			//  	-> firstly check to see if no previous other setting (in this calculation step) has met both the load and humidification requirements
+			//  			-> if so
+			//  				-> check if this setting meets the conditioning load (only) 
+			//  					-> if so
+			//  						->check to see if this setting is better at meeting the dehumidification or humidification lad than any previous setting this step.
+			//  						-> if its not, ignore it.
+			// 		  				-> if not 
+			// 		  					->check to see if any previous setting met the conditioning load 
+			// 		  						->if not:
+			// 		  							->see if this setting is better at meeting the conditioning load than any previous setting this calculation step.
+			// 		  								-> if so save as current optimal
+			// 		  								-> if its not, ignore it.
+			// 		  						-> if so: then ignore it.
+			//  			->if not, then a previous setting is better than this one by default, and so ignore it. 
+			// 9) Identify error states if the no setting meets the environmental conditions, or the supply air humidity or temperature constraints. 
+			// 10) if we met the load set operating settings to be a combination of the optimal setting at the minium required runtime fraction 
+			// 11) if we partly met the load then do the best we can and run full out in that optimal setting.
+			// 12) if we didn't even partially meet the load make sure the operational settings are just the standby mode.
+			// 13) generate summary statistics for warnings.
+
+			// REFERENCES:
+			// na
+
+			// Using/Aliasing
+			using General::RoundSigDigits;
+			// Locals
+			// SUBROUTINE ARGUMENT DEFINITIONS:
+			// The CStepInputs are defined in the CStepInputs class definition.
+			// SUBROUTINE PARAMETER DEFINITIONS:
+			// na
+			// INTERFACE BLOCK SPECIFICATIONS
+			// na
+			// DERIVED TYPE DEFINITIONS
+			// na
+
+			// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+			bool DidWeMeetLoad = false;
+			bool DidWeMeetHumidificaiton = false;
+			bool DidWePartlyMeetLoad = false;
+			int modenumber = 0;
+			int point_number = 0;
+			Real64 MsaRatio = 0;
+			Real64 OSAF = 0;
+			//Real64 Msa = 0;
+			Real64 UnscaledMsa = 0;
+			Real64 ScaledMsa = 0;
+			Real64 Mvent = 0;
+			Real64 OptimalSetting_RunFractionTotalFuel = IMPLAUSIBLE_POWER;
+			Real64 EIR;
+			Real64 ElectricalPower;
+			Real64 Tma;
+			Real64 Wma;
+			Real64 Hsa;
+			Real64 Hma;
+			Real64 TotalSystem;
+			Real64 PreviousMaxiumConditioningOutput = 0;
+			Real64 PreviousMaxiumHumidOrDehumidOutput = 0;
+			std::string ObjectID = Name.c_str();
+			int size = CurrentOperatingSettings.size();
+			CSetting empty_setting;
+			for (int i = 0; i < size; i++) 
+			{
+				CurrentOperatingSettings[i] = empty_setting;
+			}
+			if (StepIns.RHosa > 1) 
+			{ 
+				ShowSevereError("Unitary hybrid system error, required relative humidity value 0-1, called in object" + ObjectID + ".Check inputs" ); 
+				return -1;

```

Comment:
I would've sworn earlier I saw you were using try/catch, so I'd expect you to throw a meaningful severe error here, then raise an exception so that the caller can throw the final fatal.  Either that or just throw the fatal error here.

Reply:
I think I addressed this, take a look?